test/connection.test.ts

```diff
@@ -68,46 +68,222 @@ describe('Connection', function () {
       assert.calledOnce(closeSpy)
     })
 
-    it('should close all outgoing streams', async function () {
-      const spy1 = sinon.spy()
-      const spy2 = sinon.spy()
+    it('should close all outgoing streams', async function() {
+      const stream1Spy = {
+        finish: sinon.spy(),
+        end: sinon.spy(),
+        close: sinon.spy()
+      }
+      const stream2Spy = {
+        finish: sinon.spy(),
+        end: sinon.spy(),
+        close: sinon.spy()
+      }
+
       const stream1 = this.clientConn.createStream()
-      stream1.on('end', spy1)
+      stream1.on('finish', stream1Spy.finish)
+      stream1.on('end', stream1Spy.end)
+      stream1.on('close', stream1Spy.close)
       stream1.setSendMax(100)
       const stream2 = this.clientConn.createStream()
       stream2.write('hello')
-      stream2.on('finish', spy2)
+      stream2.on('finish', stream2Spy.finish)
+      stream2.on('end', stream2Spy.end)
+      stream2.on('close', stream2Spy.close)
 
       await this.clientConn.end()
 
-      assert.callCount(spy1, 1)
-      assert.callCount(spy2, 1)
+      assert.calledOnce(stream1Spy.finish)
+      assert.calledOnce(stream1Spy.end)
+      assert.calledOnce(stream1Spy.close)
+      assert.calledOnce(stream2Spy.finish)
+      assert.calledOnce(stream2Spy.end)
+      assert.calledOnce(stream2Spy.close)
+    })
+
+    it('should close all incoming streams', async function() {
+      const closeSpy = sinon.spy()
+      this.serverConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.on('close', closeSpy)
+      })
+      const stream1 = this.clientConn.createStream()
+      await stream1.setSendMax(100)
+      const stream2 = this.clientConn.createStream()
+      await stream2.write('hello')
+      await this.serverConn.end()
+
+      assert.calledTwice(closeSpy)
+    })
+
+    it.skip('should remove the stream record once one side calls end', async function() {
+      // TODO figure out how to check the stream record was removed..
+    })
+
+    it('should complete sending all data from server when end is called on server side of the connection', function() {
+      this.clientConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.on('data', (data: Buffer) => {
+          assert.equal(data.toString(), 'hello')
+        })
+      })
+      const serverStream = this.serverConn.createStream()
+      serverStream.write('hello')
+      this.serverConn.end()
+    })
+
+    it('should complete sending all money from server when end is called on server side of the connection', function() {
+      this.clientConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.setReceiveMax(1000)
+        stream.on('money', (amount) => {
+          assert.equal(amount, 200)
+        })
+      })
+      const serverStream = this.serverConn.createStream()
+      serverStream.sendTotal(100)
```

Comment:
If you're not going to await the promise it is probably better to just call `setSendMax` instead